src/checks.ts

```diff
@@ -1,44 +1,70 @@
+import assert = require("assert");
+import { makeTypesVersionsForPackageJson, TypeScriptVersion } from "definitelytyped-header-parser";
 import { pathExists } from "fs-extra";
-import * as path from "path";
+import { join as joinPaths } from "path";
 
 import { getCompilerOptions, readJson } from "./util";
 
-export async function checkPackageJson(dirPath: string): Promise<void> {
-	const pkgJsonPath = path.join(dirPath, "package.json");
+export async function checkPackageJson(
+	dirPath: string,
+	typesVersions: ReadonlyArray<TypeScriptVersion>,
+): Promise<void> {
+	const pkgJsonPath = joinPaths(dirPath, "package.json");
+	const needsTypesVersions = typesVersions.length !== 0;
 	if (!await pathExists(pkgJsonPath)) {
+		if (needsTypesVersions) {
```

Comment:
Are we requiring packages to have both the `ts3.x` version folder _and_ a `"typesVersions"` entry in a package.json file?

Reply:
Yes, unfortunately we need `package.json` to specify this so that when a DT package `a` depends on `b`, when testing `a` under ts3.1 we want `tsc` to pick up `b/ts3.1`.